baseplate/lib/live_data/zookeeper.py

```diff
@@ -56,12 +59,21 @@ def zookeeper_client_from_config(
         credentials = secrets.get_simple(path)
         auth_data.append(("digest", credentials.decode("utf8")))
 
+    # Kazoo requires different parameters depending on whether
+    # we are using gevent or not.
+    if gevent_is_patched():
+        handler = SequentialGeventHandler()
```

Comment:
This feels like something we either want or don't want all the time regardless of if gevent has patched items

Reply:
Kazoo intentionally eschews monkey patching, see the documentation below. It expects you to explicitly choose the handler yourself if you use gevent.

https://kazoo.readthedocs.io/en/latest/async_usage.html#connection-handling
```
Kazoo doesn’t rely on gevents/eventlet monkey patching and requires that you pass in the appropriate handler, the default handler is SequentialThreadingHandler
```

Kazoo docs have pretty strong guidance about not using `SequentialThreadingHandler` with gevent.

https://kazoo.readthedocs.io/en/2.6.1/api/handlers/threading.html?highlight=sequentialthreadinghandler#kazoo-sasl-handlers-threading
```
Do not use SequentialThreadingHandler with applications using asynchronous event loops (like gevent). Use the SequentialGeventHandler instead.
```

My understanding from reading the [code](https://kazoo.readthedocs.io/en/2.6.1/_modules/kazoo_sasl/handlers/threading.html#SequentialThreadingHandler) and [GitHub issues](https://github.com/python-zk/kazoo/issues/3#issuecomment-7102481) is that the Zookeeper C client library is not patched by gevent and so [spawns a separate thread](https://gevent.narkive.com/Vn0vZP9R/is-it-possible-to-execute-code-in-main-thread-when-called-from-another-thread). However, this is not well documented and we haven't been _knowingly_ hurt by it.

`SequentialThreadingHandler` _did_ seem to work okay before we patched gevent queues in #659, where we found that the handler used queue methods not available on gevent queues. This brought `SequentialGeventHandler` to our attention, which was designed to work with gevent queues from the start.

I think using an implementation built for explicitly for gevent rather than a monkey patched one is a good thing. Not only will we avoid issues with the C lib that we don't even know about, gevent code can have different semantics and performance characteristics.